slasher/rpc/detect_update_min_max_span.go

```diff
@@ -0,0 +1,132 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// Detector is an interface used to implement the slashing surround an surrounded detection
+// methods.
+type Detector interface {
+	Detect(attestationEpochSpan uint64, recorderEpochSpans *ethpb.MinMaxEpochSpan, attestationSourceEpoch uint64) uint64
+}
+
+// maxDetector is a detector used to detect surround attestations.
+type maxDetector struct{}
+
+// minDetector is a detector used to detect surrounded attestations.
+type minDetector struct{}
+
+func (d maxDetector) Detect(attestationEpochSpan uint64, recorderEpochSpans *ethpb.MinMaxEpochSpan, attestationSourceEpoch uint64) uint64 {
```

Comment:
can you put the method after the type declaration ? Having two methods with the same name is confusing one after the other

Reply:
thanks, changed